### ISSUE TITLE
NaN id loss while training with HRNetv2-w32 on custom dataset

### ISSUE THREAD

**Issue by denzelngo** [ifzhang/FairMOT#205]
Hi @ifzhang 
Thanks for sharing the great work that you have done!

I have tried training the model with HRNetv2 backbone on my custom dataset which is a vehicle based dataset. I have followed the data preprocessing instruction in #55 . I use this command line to run the training:
`python train.py mot --exp_id all_hrnet --gpus 2 --batch_size 3 --arch 'hrnet_32' --lr 1e-5`
During the training, I have the problem that **the value of loss adn id_loss are NaN**.

![image](https://user-images.githubusercontent.com/47851073/88534615-bf898d00-d008-11ea-91ba-69a904c3d5b0.png)

Thanking you in anticipation!

**Comment by ifzhang** [comment]
It's a little bit strange that the id loss seems normal in the beginning. You may try to uncomment this line to train the parameters of the id_classifier:
https://github.com/ifzhang/FairMOT/blob/ca78d3f01d1953c3d72ee8b01f24a5ddec2689c3/src/lib/trains/base_trainer.py#L30

**Comment by denzelngo** [comment]
I have dealt with the problem. I found that when the input image in which there is no object, which means **id_target is empty** and **id_output** **is also empty**, **the id_loss is NaN**. Therefore, I add some lines of code to make sure that `if len(id_target)>0 and len(id_output)>0: 
id_loss += self.IDLoss(id_output, id_target)` 
Thank you for your response.

**Comment by denzelngo** [comment]
by the way, can you explain for me more clearly about some parameters of the input data (batch['reg_mask'], batch['reg'] and batch['ind']? As I've read in the #109 , you've said that batch['ind'] contains all ground truth center locations but can you explain why you calculate it with this code:
`ind[k] = ct_int[1] * output_w + ct_int[0]`
I'm looking forward to your answer.
